### ISSUE TITLE
Documentation: how it works

### ISSUE THREAD

**Issue by shykes** [dagger/dagger#1784]
There should be a documentation page talking about how Dagger works under the hood. An architecture diagram, core technology, etc.

For example this diagram is from https://sigstore.dev/how-it-works:

![Uploading 37806784-F7C7-42C4-9460-9EA33E8E1646.jpeg…]()

**Comment by sipsma** [comment]
I'm imagining a high-level approach where we split this into 3 sections:
1. What is BuildKit
   * the `BuildKit Crash Course` section of [my blog post here](https://www.docker.com/blog/mergediff-building-dags-more-efficiently-and-elegantly/) may be a good reference for how to explain it quickly (though probably don't use Dockerfile as a central example to avoid confusion)
   * Include a diagram of an LLB DAGs
2. What is `cue/flow`
   * This will reference our current [What is CUE?](https://docs.dagger.io/1215/what-is-cue/) page, but focus more on how `cue/flow` enables us to create DAGs of tasks that are defined with CUE but implemented using Go
   * Include a diagram of CUE task DAGs
3. How do we bridge from the `cue/flow` DAG -> BuildKit LLB DAG
   * Include diagram showing the bridge from the `cue/flow` DAG of the previous example to the LLB DAG from the earlier example.

Basically, the approach here is to progressively introduce concepts and then conclude with how they all connect together. It should be something like a peek under the hood but also doesn't require going into too low-level of detail.

I'll start with creating a skeleton of this approach and placeholder diagrams/explanations.

**Comment by sipsma** [comment]
Moved this back from in progress for now as there are some emergent high priorities (which may also impact what we write in these docs). Will reschedule once there is bandwidth again.

**Comment by mircubed** [comment]
removing @sipsma as assignee since he is working on a few other things right now. @jpadams, Erik offered to help answer questions around buildkit (if needed) whenever we get someone else to work on this.
